ydb/library/actors/wilson/wilson_uploader.cpp

```diff
@@ -179,15 +266,20 @@ namespace NWilson {
             template<typename T>
             void ScheduleWakeup(T&& deadline) {
                 if (!WakeupScheduled) {
-                    TActivationContext::Schedule(deadline, new IEventHandle(TEvents::TSystem::Wakeup, 0, SelfId(), {},
-                        nullptr, 0));
+                    TActivationContext::Schedule(deadline,
+                                                 new IEventHandle(SelfId(), {}, new TEvents::TEvWakeup));
                     WakeupScheduled = true;
                 }
             }
 
-            void HandleWakeup() {
-                Y_ABORT_UNLESS(WakeupScheduled);
-                WakeupScheduled = false;
+            void HandleWakeup(TEvents::TEvWakeup::TPtr& ev) {
+                const auto tag = ev->Get()->Tag;
+                if (tag == SendBatchId) {
+                    CompleteCurrentBatch();
```

Comment:
I thought introducing a new event type for this would've been overkill. Maybe it isn't